### ISSUE TITLE
GitHub: Cannot read properties of undefined (reading 'find')

### ISSUE THREAD

**Issue by andrearagonvturb** [TeamCodeStream/codestream#1435]
- CodeStream Extension version:
- vscode version 1.77.1
- Linux, ubuntu 20.04

Since the first time I started codestream, i've been seeing the following message when trying to access the my pull requests:

Request codestream/provider/generic failed with message: Cannot read properties of undefined (reading 'find')

I see the same error in other moments, when I try to open a new PR, I see a similar message:

GitHub: Cannot read properties of undefined (reading 'find')

In the CodeStream agent output, I find the following error log:

[Error - 8:48:08 AM] [2023-04-17 11:48:08:270] GitHub: handleProviderError — {"request":{"providerId":"github*com","remote":"//github.com/vturb/analytics-process-by-query"}}
TypeError: Cannot read properties of undefined (reading 'find')
TypeError: Cannot read properties of undefined (reading 'find')
    at so._isSuppressedException (/home/aragon/.vscode/extensions/codestream.codestream-14.18.0/src/providers/thirdPartyIssueProviderBase.ts:265:27)
    at so.query (/home/aragon/.vscode/extensions/codestream.codestream-14.18.0/src/providers/github.ts:294:24)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at so.getRepoInfo (/home/aragon/.vscode/extensions/codestream.codestream-14.18.0/src/providers/github.ts:1057:21)
[2023-04-17 11:48:08:271] [2d0] GitHubProvider.getRepoInfo completed • 555 ms
[2023-04-17 11:48:08:271] [2cc] ReviewsManager.checkPullRequestPreconditions completed • 616 ms
[2023-04-17 11:48:18:774] [2d1] CodeStreamApiProvider.updatePreferences
[2023-04-17 11:48:18:783] github getMyPullRequests {"prQueries":[{"providerId":"github*com","name":"Waiting on my Review","query":"is:pr is:open review-requested:@me","hidden":true},{"providerId":"github*com","name":"Assigned to Me","query":"is:pr is:open assignee:@me","hidden":false},{"providerId":"github*com","name":"Created by Me","query":"is:pr is:open author:@me","hidden":true},{"providerId":"github*com","name":"Recent","query":"recent","hidden":true}],"isOpen":false,"force":true}
[2023-04-17 11:48:18:784] [2d2] ThirdPartyProviderRegistry.getProviderDefaultPullRequestQueries
[2023-04-17 11:48:18:784] [2d2] ThirdPartyProviderRegistry.getProviderDefaultPullRequestQueries completed • 0 ms
[2023-04-17 11:48:18:784] Getting login for providerId: github*com
[Warn  - 8:48:19 AM] [2023-04-17 11:48:19:321] GitHub query caught (elapsed=536.8474690020084ms): — {"response":{"message":"Bad credentials","documentation_url":"https://docs.github.com/graphql","status":401,"headers":{}},"request":{"query":"query {\n\t\t\t\t\t\tviewer {\n\t\t\t\t\t\t  id\n\t\t\t\t\t\t\tlogin\n\t\t\t\t\t\t}\n\t\t\t\t\t}"}}
Ignoring duplicate error {
  key: 'mZFLkyvTelC5g8XnyQrpOw==',
  message: "Cannot read properties of undefined (reading 'find')"
}
[Error - 8:48:19 AM] [2023-04-17 11:48:19:322] executeMethod failed — {"method":"getMyPullRequests"}
TypeError: Cannot read properties of undefined (reading 'find')
TypeError: Cannot read properties of undefined (reading 'find')
    at so._isSuppressedException (/home/aragon/.vscode/extensions/codestream.codestream-14.18.0/src/providers/thirdPartyIssueProviderBase.ts:265:27)
    at so.query (/home/aragon/.vscode/extensions/codestream.codestream-14.18.0/src/providers/github.ts:294:24)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at so.getGithubLogin (/home/aragon/.vscode/extensions/codestream.codestream-14.18.0/src/providers/github.ts:5975:20)
    at so.getMyPullRequests (/home/aragon/.vscode/extensions/codestream.codestream-14.18.0/src/providers/github.ts:3113:23)
    at li.executeMethod (/home/aragon/.vscode/extensions/codestream.codestream-14.18.0/src/providers/registry.ts:674:13)
[2023-04-17 11:48:19:543] [2d4] CodeStreamAgent.sendNotification(codestream/refreshMaintenancePoll)
[2023-04-17 11:48:19:544] [2d4] CodeStreamAgent.sendNotification(codestream/refreshMaintenancePoll) completed • 0 ms
[2023-04-17 11:48:19:545] API(1050e80b-e475-47fd-9c75-70b8f4788db3): Completed PUT /preferences body={"sidebarPanes":{"open-pull-requests":{"collapsed":false}}} • 770 ms
[2023-04-17 11:48:19:546] [2d7] CodeStreamAgent.sendNotification(codestream/didChangeData:users)
[2023-04-17 11:48:19:547] [2d7] CodeStreamAgent.sendNotification(codestream/didChangeData:users) completed • 0 ms
[2023-04-17 11:48:19:547] [2d8] CodeStreamAgent.sendNotification(codestream/didChangeData:preferences)
[2023-04-17 11:48:19:548] [2d8] CodeStreamAgent.sendNotification(codestream/didChangeData:preferences) completed • 0 ms
[2023-04-17 11:48:19:548] [2d1] CodeStreamApiProvider.updatePreferences completed • 773 ms
[2023-04-17 11:48:19:552] BROADCASTER: PUBNUB: Message received on user-63e177ddddb972d9d10df57a at 1681732099478
[2023-04-17 11:48:19:552] BROADCASTER: Broadcaster message received at: 1681732099478
[2023-04-17 11:48:19:552] BROADCASTER: _lastMessageReceivedAt updated to 1681732099478
[2023-04-17 11:48:52:920] [2dc] GitHubProvider.getBoards
[2023-04-17 11:48:52:920] [2dd] GitService.getRepoRemotes
[2023-04-17 11:48:52:920] [2dd] GitService.getRepoRemotes returned [ssh://git@github.com:vturb/analytics-process-by-query.git] • 0 ms
[2023-04-17 11:48:52:922] [2de] GitHubProvider.getCards
[2023-04-17 11:48:53:324] GitHub: FAILED(0x) GET https://api.github.com/search/issues?q=archived%3Afalse%20is%3Aissue%20is%3Aopen%20assignee%3A%40me&sort=updated • 401 ms
[Warn  - 8:48:53 AM] [2023-04-17 11:48:53:324] github query caught: — {"info":{},"name":"InternalError"}
[2023-04-17 11:48:53:324] Error from GitHub:  — {
    "info": {},
    "name": "InternalError"
}
[2023-04-17 11:48:53:324] [2de] GitHubProvider.getCards completed • 401 ms
[2023-04-17 11:48:53:341] GitHub: FAILED(0x) GET https://api.github.com/repos/vturb/analytics-process-by-query • 420 ms
Ignoring duplicate error { key: '0K7ztZr/eU1T03bd4HxVFw==', message: 'Unauthorized' }
[Warn  - 8:48:53 AM] [2023-04-17 11:48:53:341] github query caught: — {"info":{},"name":"InternalError"}
[Error - 8:48:53 AM] [2023-04-17 11:48:53:342] so.handleErrorResponse
InternalError: Unauthorized
InternalError: Unauthorized
    at so.handleErrorResponse (/home/aragon/.vscode/extensions/codestream.codestream-14.18.0/src/providers/thirdPartyProviderBase.ts:493:11)
    at so.fetch (/home/aragon/.vscode/extensions/codestream.codestream-14.18.0/src/providers/thirdPartyProviderBase.ts:463:30)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at so.get (/home/aragon/.vscode/extensions/codestream.codestream-14.18.0/src/providers/github.ts:392:15)
    at so.restGet (/home/aragon/.vscode/extensions/codestream.codestream-14.18.0/src/providers/github.ts:486:20)
    at getOpenedRepos (/home/aragon/.vscode/extensions/codestream.codestream-14.18.0/src/providers/provider.ts:345:24)
    at so.getBoards (/home/aragon/.vscode/extensions/codestream.codestream-14.18.0/src/providers/github.ts:504:24)
[2023-04-17 11:48:53:742] GitHub: FAILED(0x) GET https://api.github.com/user/repos • 399 ms
Ignoring duplicate error { key: '0K7ztZr/eU1T03bd4HxVFw==', message: 'Unauthorized' }
[Warn  - 8:48:53 AM] [2023-04-17 11:48:53:742] github query caught: — {"info":{},"name":"InternalError"}
[Error - 8:48:53 AM] [2023-04-17 11:48:53:743] so.handleErrorResponse
InternalError: Unauthorized
InternalError: Unauthorized
    at so.handleErrorResponse (/home/aragon/.vscode/extensions/codestream.codestream-14.18.0/src/providers/thirdPartyProviderBase.ts:493:11)
    at so.fetch (/home/aragon/.vscode/extensions/codestream.codestream-14.18.0/src/providers/thirdPartyProviderBase.ts:463:30)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at so.get (/home/aragon/.vscode/extensions/codestream.codestream-14.18.0/src/providers/github.ts:392:15)
    at so.restGet (/home/aragon/.vscode/extensions/codestream.codestream-14.18.0/src/providers/github.ts:486:20)
    at so.getBoards (/home/aragon/.vscode/extensions/codestream.codestream-14.18.0/src/providers/github.ts:525:26)
[2023-04-17 11:48:53:743] [2dc] GitHubProvider.getBoards completed • 823 ms
[2023-04-17 11:49:18:317] github getMyPullRequests {"prQueries":[{"providerId":"github*com","name":"Waiting on my Review","query":"is:pr is:open review-requested:@me","hidden":true},{"providerId":"github*com","name":"Assigned to Me","query":"is:pr is:open assignee:@me","hidden":false},{"providerId":"github*com","name":"Created by Me","query":"is:pr is:open author:@me","hidden":true},{"providerId":"github*com","name":"Recent","query":"recent","hidden":true}],"isOpen":false,"force":true}
[2023-04-17 11:49:18:317] [2df] ThirdPartyProviderRegistry.getProviderDefaultPullRequestQueries
[2023-04-17 11:49:18:317] [2df] ThirdPartyProviderRegistry.getProviderDefaultPullRequestQueries completed • 0 ms
[2023-04-17 11:49:18:317] Getting login for providerId: github*com
[Warn  - 8:49:18 AM] [2023-04-17 11:49:18:664] GitHub query caught (elapsed=346.6418250016868ms): — {"response":{"message":"Bad credentials","documentation_url":"https://docs.github.com/graphql","status":401,"headers":{}},"request":{"query":"query {\n\t\t\t\t\t\tviewer {\n\t\t\t\t\t\t  id\n\t\t\t\t\t\t\tlogin\n\t\t\t\t\t\t}\n\t\t\t\t\t}"}}
Ignoring duplicate error {
  key: 'mZFLkyvTelC5g8XnyQrpOw==',
  message: "Cannot read properties of undefined (reading 'find')"
}

![image](https://user-images.githubusercontent.com/107865458/232477264-e25cd13b-20b3-417b-80a8-54448ea59f5d.png)

Steps to Reproduce:
I don't know how to reproduce the issue.

Does anyone here know how to fix it?

**Comment by planteater** [comment]
@andrearagonvturb - First thing I would check is to make sure the personal access token you used to connect with has the appropriate scopes:

repo,read:user,user:email,notifications,read:org

**Comment by andrearagonvturb** [comment]
Hi! Thank you so much for the help! That was the issue! The logs where so large that I could not find a small message:

{"message":"Bad credentials","documentation_url":"[https://docs.github.com/graphql","status":401,"headers":{}},"request":

This was happening because my github access token was expired.
